docs/orchestrating-elastic-stack-applications/stack-config-policy.asciidoc

```diff
@@ -0,0 +1,236 @@
+:page_id: stack-config-policy
+ifdef::env-github[]
+****
+link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}.html[View this document on the Elastic website]
+****
+endif::[]
+[id="{p}-{page_id}"]
+= Elastic Stack configuration policies
+
+NOTE: This requires a valid Enterprise license or Enterprise trial license. Check <<{p}-licensing,the license documentation>> for more details about managing licenses.
+
+Starting from ECK `2.6.0` and Elasticsearch `8.6.0`, Elastic Stack configuration policies let you configure in a declarative way, for multiple clusters, the following settings:
+
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html[Cluster Settings]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/put-snapshot-repo-api.html[Snapshot Repositories]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-api-put-policy.html[Snapshot Lifecycle Policies]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/put-pipeline-api.html[Ingest pipelines]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-put-lifecycle.html[Index Lifecycle Policies]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-template.html[Index templates]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-component-template.html[Components templates]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-role-mapping.html[Role mappings]
+
+A policy can be applied to one or more Elasticsearch clusters in any namespace managed by the ECK operator.
+
+Configuration policy settings applied by the ECK operator are immutable through the Elasticsearch REST API.
+
+It is currently not allowed to configure an Elasticsearch cluster with more than one policy.
+
+[float]
+[id="{p}-{page_id}-definition"]
+== Define Elastic Stack configuration policies
+
+Elastic Stack configuration policies can be defined in a `StackConfigPolicy` resource. Each `StackConfigPolicy` must have the following fields:
+
+* `name` is a unique name used to identify the policy.
+* `spec.elasticsearch` describes the settings to configure and at least one setting must be defined. Each of the following fields is a named entity map:
```

Comment:
"named entity map" is not self-explanatory. Also I don't think it is true for the `clusterSettings`?

Reply:
True. What do you thing of:

> Each of the following fields except `clusterSettings` is an associative array where the keys are arbitrary names and the values are the definitions: